lab08-jacob/route/user-route.js

```diff
@@ -0,0 +1,64 @@
+const storage = require('../lib/storage.js');
+const response = require('../lib/response.js');
+const User = require('../model/resource.js');
+
+module.exports = function(router) {
+  router.get('/api/users', function(req, res) {
+    if (req.url.query.id) {
+      storage.fetchItem('users', req.url.query.id)
+      .then(user => {
+        response.sendJSON(res, 200, user);
+      })
+      .catch(err => {
+        console.error(err);
+        response.sendText(res, 404, 'not found');
+      });
+      return;
+    }
+    response.sendText(res, 400, 'bad request');
+  });
+
+  router.post('/api/users', function(req, res) {
+    if(Object.keys(req.body).length === 0) {
+      res.writeHead(400, {'Content-Type': 'text/plain',
+      });
+      res.write('include body');
+      res.end();
+      return;
+    }
+    try {
+      var user = new User(req.body.username);
+      storage.createItem('users', user);
+      res.writeHead(200, {
+        'Content-Type': 'application/json',
+      });
+      res.write(JSON.stringify(user));
+      res.end();
+    } catch (err) {
+      console.error(err);
+      res.writehead(400, {
+        'Content-Type': 'text/plain',
+      });
+      res.write('you can\'t post that smut!');
+      res.end();
+    }
+  });
+
+  router.delete('/api/users', function(req, res) {
+    try {
+      var user = (req.url.query.id);
+      storage.deleteUser('users', user);
+      res.writeHead(204, {
+        'Content-Type': 'text/plain',
+      });
+      res.end(); //normally I would log 'user deleted' or something.
```

Comment:
Check the actual spec for an HTTP Delete method. From the perspective of an API, you shouldn't need to log anything since your http response, specifically the status code in this case, should tell the origin of the request that the delete was successful. Logs are just for the developer(s), and generally shouldn't be used for formal lines of communication.